src/server/db_slice.cc

```diff
@@ -1076,15 +1077,19 @@ DbSlice::ItAndExp DbSlice::ExpireIfNeeded(const Context& cntx, PrimeIterator it)
 
   auto expire_it = db->expire.Find(it->first);
 
-  CHECK(IsValid(expire_it));
-
-  // TODO: to employ multi-generation update of expire-base and the underlying values.
-  time_t expire_time = ExpireTime(expire_it);
-
-  // Never do expiration on replica or if expiration is disabled.
-  if (time_t(cntx.time_now_ms) < expire_time || owner_->IsReplica() || !expire_allowed_)
-    return {it, expire_it};
+  if (IsValid(expire_it)) {
+    // TODO: to employ multi-generation update of expire-base and the underlying values.
+    time_t expire_time = ExpireTime(expire_it);
 
+    // Never do expiration on replica or if expiration is disabled.
+    if (time_t(cntx.time_now_ms) < expire_time || owner_->IsReplica() || !expire_allowed_)
+      return {it, expire_it};
+  } else {
+    LOG(ERROR) << "Internal error, entry " << it->first.ToString()
+               << " not found in expire table, db_index: " << cntx.db_index
+               << ", expire table size: " << db->expire.size()
+               << ", prime table size: " << db->prime.size() << util::fb2::GetStacktrace();
```

Comment:
missing return

Reply:
it's not missing. I treat missing expire as "EXPIRED" and continue deleting the inconsistent entry that has expire bit set on it.

Reply:
Ok I would do that after check that it->second.HasExpire()
might we have a bug that we call this functions when the entry does not have expire bit